### PR TITLE
Add docs for pagers and pollers

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
- [ ] The purpose of this PR is explained in this or a referenced issue.
- [ ] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [ ] Tests are included and/or updated for code changes.
- [ ] Updates to [CHANGELOG.md][] are included.
- [ ] MIT license headers are included in each file.

[Azure/autorest.go]: https://github.com/Azure/autorest.go
[CHANGELOG.md]: https://github.com/Azure/azure-sdk-for-go/blob/main/CHANGELOG.md

### CODE REVIEW

**Comment by jhendrixMSFT** [comment]
Fixes https://github.com/Azure/azure-sdk-for-go/issues/3042

File: sdk/azcore/doc.go [hunk 1]
```diff
@@ -6,7 +6,7 @@
 // license that can be found in the LICENSE file.
 
 /*
-Package azcore implements an HTTP request/response middleware pipeline.
+Package azcore implements an HTTP request/response middleware pipeline used by the Azure SDKs.
```
Comment (chlowell) [inline sdk/azcore/doc.go:9]: "Azure SDK**s**" used to be a messaging bugbear--there's only one SDK to rule them all!--but maybe not anymore; this splits the difference:
```suggestion
Package azcore implements an HTTP request/response middleware pipeline used by Azure SDK clients.
```

File: sdk/azcore/doc.go [hunk 2]
```diff
@@ -177,5 +177,81 @@ a callback that writes to the desired location.  Any custom logging implementati
...
+   *  Canceled - the LRO was canceled
+
+LROs can be identified by their Begin* prefix and their return type of *runtime.Poller[T].
+
+   func (c *WidgetClient) BeginCreateOrUpdate(ctx context.Context, w Widget, o *Options) (*runtime.Poller[Response], error)
+
+When a call to WidgetClient.BeginCreateOrUpdate() returns a nil error, it means that the LRO has started.
+It does _not_ mean that the widget has been created or updated (or failed to be created/updated).
+
+The *runtime.Poller[T] provides APIs for determining the state of the LRO.  To wait for the LRO complete,
```
Comment (chlowell) [inline sdk/azcore/doc.go:219]:
```suggestion
The *runtime.Poller[T] provides APIs for determining the state of the LRO.  To wait for the LRO to complete,
```

File: sdk/azcore/doc.go [hunk 3]
```diff
@@ -177,5 +177,81 @@ a callback that writes to the desired location.  Any custom logging implementati
...
+   token, err := poller.ResumeToken()
+   // handle error
+
+Note that a token can only be obtained for a poller that's in a non-terminal state.  Also note that any subsequent calls
+to poller.Poll() might change the poller's state.  In this case, a new token should be created.
+
+After the token has been obtained, it can be used to recreate an instance of the originating poller.
+
+   poller, err := widgetClient.BeginCreateOrUpdate(nil, Widget{}, &Options{
+      ResumeToken, token,
```
Comment (chlowell) [inline sdk/azcore/doc.go:249]:
```suggestion
      ResumeToken: token,
```

**Review by chlowell — APPROVED** [review]

File: sdk/azcore/doc.go (hunk 3 above)
Comment (jhendrixMSFT) [reply sdk/azcore/doc.go]: fixed

File: sdk/azcore/doc.go (hunk 2 above)
Comment (jhendrixMSFT) [reply sdk/azcore/doc.go]: fixed

File: sdk/azcore/doc.go (hunk 1 above)
Comment (jhendrixMSFT) [reply sdk/azcore/doc.go]: fixed

**Pull request merged**